service/src/test/java/uk/gov/hmcts/reform/fpl/handlers/NotificationHandlerTest.java

```diff
@@ -67,9 +77,65 @@
     @Mock
     private GatekeeperEmailContentProvider gatekeeperEmailContentProvider;
 
+    @Mock
+    private C2UploadedEmailContentProvider c2UploadedEmailContentProvider;
+
+    @Mock
+    private IdamApi idamApi;
+
     @InjectMocks
     private NotificationHandler notificationHandler;
 
+    @Nested
+    class C2UploadedNotificationChecks {
+        final String subjectLine = "Lastname, SACCCCCCCC5676576567";
+        final Map<String, Object> parameters = ImmutableMap.<String, Object>builder()
+            .put("subjectLine", subjectLine)
+            .put("hearingDetailsCallout", subjectLine)
+            .put("reference", "12345")
+            .put("caseUrl", "null/case/" + JURISDICTION + "/" + CASE_TYPE + "/12345")
+            .build();
+
+        @Test
+        void shouldNotNotifyHmctsAdminOnC2Upload() throws IOException, NotificationClientException {
+            given(idamApi.retrieveUserDetails(AUTH_TOKEN))
+                .willReturn(new UserDetails("1", "hmcts-admin@test.com",
+                    "Hmcts", "Test", HMCTS_ADMIN.getRoles()));
+
+            given(c2UploadedEmailContentProvider.buildC2UploadNotification(callbackRequest().getCaseDetails()))
```

Comment:
this mock can be extracted to a before and removed from the tests

Reply:
@Gogetter

Reply:
@swalker125 yes it can but they use two different emails and different UserRole